code/modules/food_and_drinks/kitchen_machinery/cooker.dm

```diff
@@ -147,6 +197,8 @@
 			else
 				L.death()
 		break
+	if(istype(I, /obj/item/grab))
+		return special_attack_grab(I, user)
```

Comment:
Also is this necessary if `checkValid` is going to already run this check?

Reply:
Moved the sanity checking logic out and left only that in checkValid, so now attackby is the only place it'd trigger.